### ISSUE TITLE
mbcopy.cc does not build with Visual Studio

### ISSUE THREAD

**Issue by joa-quim** [dwcaress/MB-System#1081]
```
   Creating library mbcopy.lib and object mbcopy.exp
LINK : warning LNK4098: defaultlib 'MSVCRTD' conflicts with use of other libs; use /NODEFAULTLIB:library
mbcopy.obj : error LNK2019: unresolved external symbol "double * angles_EM1000_104_07_MS_48_FAIS" (?angles_EM1000_104_07_MS_48_FAIS@@3PANA) referenced in function "int __cdecl mbcopy_simrad_to_simrad2(int,struct mbsys_simrad_struct *,struct mbsys_simrad2_struct *,int *)" (?mbcopy_simrad_to_simrad2@@YAHHPEAUmbsys_simrad_struct@@PEAUmbsys_simrad2_struct@@PEAH@Z)
mbcopy.obj : error LNK2019: unresolved external symbol "double * angles_EM1000_120_07_MS_48_FAIS" (?angles_EM1000_120_07_MS_48_FAIS@@3PANA) referenced in function "int __cdecl mbcopy_simrad_to_simrad2(int,struct mbsys_simrad_struct *,struct mbsys_simrad2_struct *,int *)" (?mbcopy_simrad_to_simrad2@@YAHHPEAUmbsys_simrad_struct@@PEAUmbsys_simrad2_struct@@PEAH@Z)
mbcopy.obj : error LNK2019: unresolved external symbol "double * angles_EM1000_128_02_MS_60_FAIS" (?angles_EM1000_128_02_MS_60_FAIS@@3PANA) referenced in function "int __cdecl mbcopy_simrad_to_simrad2(int,struct mbsys_simrad_struct *,struct mbsys_simrad2_struct *,int *)" (?mbcopy_simrad_to_simrad2@@YAHHPEAUmbsys_simrad_struct@@PEAUmbsys_simrad2_struct@@PEAH@Z)
mbcopy.obj : error LNK2019: unresolved external symbol "double * angles_EM1000_140_02_MS_60_FAIS" (?angles_EM1000_140_02_MS_60_FAIS@@3PANA) referenced in function "int __cdecl mbcopy_simrad_to_simrad2(int,struct mbsys_simrad_struct *,struct mbsys_simrad2_struct *,int *)" (?mbcopy_simrad_to_simrad2@@YAHHPEAUmbsys_simrad_struct@@PEAUmbsys_simrad2_struct@@PEAH@Z)
...

reverting back to C version and builds fine.
(and it's not about the warning ``LINK : warning LNK4098: defaultlib 'MSVCRTD' conflicts with use of other libs; use /NODEFAULTLIB:library`` , I get the same in C)
)
```

**Comment by dwcaress** [comment]
I've attempted to fix this with commit #1083.

**Comment by joa-quim** [comment]
Thanks. It now builds, but also neither ``mbset`` or ``mbmakeplatform`` build because of the *too many nested if-else* problem. See
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1061?view=vs-2019

**Comment by dwcaress** [comment]
Do case statements have a similar limitation?

**Comment by joa-quim** [comment]
I don't think they have because they are kind of disguised gotos. In GMT we have also a some functions (``gmtlib_setparameter()`` for example) with lots of cases and I never had a complain like this.

**Comment by dwcaress** [comment]
I've attempted to fix mbset and mbmakeplatform for you with #1086. I tried two approaches - a switch case statement for mbset and non-nested, separate if statements for mbmakeplatform.

**Comment by joa-quim** [comment]
Thanks, that fixed this issue.
